modules/oe_content_persistent/src/ContentUuidResolver.php

```diff
@@ -0,0 +1,224 @@
+      }
+      else {
+        $this->cache->set($this->cacheKey, reset($this->lookupMap), $this->getRequestTime() + $twenty_four_hours);
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAliasByUuid(string $uuid, string $langcode = NULL): ?string {
+    $langcode = $langcode ?: $this->languageManager->getCurrentLanguage(LanguageInterface::TYPE_URL)->getId();
+
+    // Here we are trying to use static cache.
+    if (!empty($this->lookupMap[$uuid])) {
+      return $this->lookupMap[$uuid];
+    }
+
+    // Using cache key initialized in controller
+    // (if we use service inside controller).
+    if ($this->cacheKey) {
+      if ($cached = $this->cache->get($this->cacheKey)) {
+        // Used for bubble up cache tags to page cache level.
+        $this->addCacheTags($cached->tags);
+        return $cached->data;
+      }
+      else {
+        // Informing about cache writing on kernel termination.
+        // @see \Drupal\oe_content_persistent\EventSubscriber\UuidPathSubscriber and $this->writeCache().
+        $this->cacheNeedsWriting = TRUE;
+      }
+    }
+    // Reuse previously cached alias of uuid,
+    // even if we don't use this service with controller.
+    elseif ($cached = $this->cache->get($this->getCachePrefixKey() . $uuid)) {
+      return $cached->data;
+    }
+
+    // Try to retrieve entities from supported storages.
+    foreach ($this->supportedStorages as $storage_type) {
+      $storage = $this->entityTypeManager->getStorage($storage_type);
+      // Retrieving correct entity by uuid as we can't get entity internal url
+      // without loading full entity.
+      $entities = $storage->loadByProperties(['uuid' => $uuid]);
+      if (empty($entities)) {
+        return NULL;
+      }
+
+      $entity = reset($entities);
+      if ($entity instanceof EntityInterface) {
+        $alias = $this->aliasManager->getAliasByPath('/' . $entity->toUrl()->getInternalPath(), $langcode);
+        // Normalize url with related parts like langcode prefix and base url.
+        $this->lookupMap[$uuid] = Url::fromUserInput($alias)->toString();
+        $this->uuidCacheTags[$uuid] = $entity->getCacheTags();
+        // Used for bubble up cache tags to page cache level.
+        $this->addCacheableDependency($entity);
+        break;
+      }
+    }
+
+    return $this->lookupMap[$uuid] ?? NULL;
+  }
+
+  /**
+   * Return prefix for cache key.
+   */
+  protected function getCachePrefixKey(): string {
+    $langcode = $this->languageManager->getCurrentLanguage(LanguageInterface::TYPE_URL)->getId();
+    return 'content_uuid:' . $langcode . ':';
+  }
+
+  /**
+   * Wrapper method for REQUEST_TIME constant.
+   *
+   * @return int
+   *   Return current timestamp.
+   */
+  protected function getRequestTime(): int {
+    return defined('REQUEST_TIME') ? REQUEST_TIME : (int) $_SERVER['REQUEST_TIME'];
```

Comment:
\Drupal::time()->getRequestTime()

Reply:
Anyway for now we have decided to remove caching.